xmake/rules/c++/modules/build_modulefiles.lua

```diff
@@ -103,11 +103,11 @@ function _build_modulefiles_msvc(target, sourcebatch, opt)
     for _, sourcefile in ipairs(sourcebatch.sourcefiles) do
         local objectfile = target:objectfile(sourcefile)
         local dependfile = target:dependfile(objectfile)
-        local modulefile = objectfile .. ".pcm"
+        local modulefile = objectfile .. ".ifc"
 
         -- compile module file to *.pcm
         local singlebatch = {sourcekind = "cxx", sourcefiles = {sourcefile}, objectfiles = {objectfile}, dependfiles = {dependfile}}
-        opt.configs.cxxflags = {"/experimental:module /module:interface /module:output " .. os.args(modulefile), "/TP"}
+        opt.configs.cxxflags = {"/experimental:module /interface /ifcOutput " .. os.args(modulefile), "/TP"}
```

Comment:
Is this change compatible with lower versions vs?

Reply:
I guess VS2017 may not support c++ modules, and I don't have one to test.
And although I have a VS2019 build tools, xmake doesn't use that even I have configured xmake to it. And it keeps changing my config from --vs=2019 to --vs=2022. May I know how you treat with vswhere.exe?

Reply:
Well, You're right, in vs 2015, 2017, 2019, it's still "/experimental:module" but not "/module". Then, how to judge the version of vs?

Reply:
use has_flags to detect it. it should have been checked. 

https://github.com/xmake-io/xmake/blob/bc582fd4bdaee3add88a4e15f1764c7e11eec41c/xmake/rules/c%2B%2B/modules/build_modulefiles.lua#L137